libbeat/publisher/pipeline/controller.go

```diff
@@ -169,45 +179,99 @@ func (c *outputController) Reload(
 	return nil
 }
 
+// queueProducer creates a queue producer with the given config, blocking
+// until the queue is created if it does not yet exist.
 func (c *outputController) queueProducer(config queue.ProducerConfig) queue.Producer {
-	return c.queue.Producer(config)
+	c.queueLock.Lock()
+	if c.queue != nil {
+		defer c.queueLock.Unlock()
```

Comment:
Not in this case (I've added a more detailed comment, sorry I meant to include one already) -- because in the other branch, where the queue doesn't exist, we need to manually unlock so we don't hold the mutex while waiting for the queue to be created.